testgrid/cmd/configurator/prow_test.go

```diff
@@ -571,6 +570,231 @@ func Test_applySingleProwjobAnnotation_WithDefaults(t *testing.T) {
+								Name: "ci-kubernetes-e2e",
+							},
+						},
+						prowConfig.Postsubmit{
+							JobBase: prowConfig.JobBase{
+								Name: "ci-kubernetes-unit",
+							},
+						},
+					},
+					"containerd/cri": {
+						prowConfig.Postsubmit{
+							JobBase: prowConfig.JobBase{
+								Name: "pull-cri-containerd-build",
+							},
+						},
+					},
+				},
+				Periodics: []prowConfig.Periodic{
+					{
+						JobBase: prowConfig.JobBase{
+							Name: "ESPv2-continuous-build",
+						},
+					},
+					{
+						JobBase: prowConfig.JobBase{
+							Name: "everlast-bump",
+						},
+					},
+					{
+						JobBase: prowConfig.JobBase{
+							Name: "ci-oss-test-infra-autobump-prow",
+						},
+					},
+				},
+			},
+			expectedJobConfig: prowConfig.JobConfig{
+				PresubmitsStatic: map[string][]prowConfig.Presubmit{
+					"helm/helm": {
+						prowConfig.Presubmit{
+							JobBase: prowConfig.JobBase{
+								Name: "pull-test-go",
+							},
+						},
+					},
+					"istio/proxy": {
+						prowConfig.Presubmit{
+							JobBase: prowConfig.JobBase{
+								Name: "gen_check_master",
+							},
+						},
+						prowConfig.Presubmit{
+							JobBase: prowConfig.JobBase{
+								Name: "lint_master",
+							},
+						},
+						prowConfig.Presubmit{
+							JobBase: prowConfig.JobBase{
+								Name: "lint_release-1.5",
+							},
+						},
+					},
+					"kubernetes/test-infra": {
+						prowConfig.Presubmit{
+							JobBase: prowConfig.JobBase{
+								Name: "pull-test-bazel",
+							},
+						},
+					},
+				},
+				PostsubmitsStatic: map[string][]prowConfig.Postsubmit{
+					"GoogleCloudPlatform/oss-test-infra": {
+						prowConfig.Postsubmit{
+							JobBase: prowConfig.JobBase{
+								Name: "pull-test-infra-go-test",
+							},
+						},
+					},
+					"containerd/cri": {
+						prowConfig.Postsubmit{
+							JobBase: prowConfig.JobBase{
+								Name: "pull-cri-containerd-build",
+							},
+						},
+					},
+					"kubernetes/kubernetes": {
+						prowConfig.Postsubmit{
+							JobBase: prowConfig.JobBase{
+								Name: "ci-kubernetes-e2e",
+							},
+						},
+						prowConfig.Postsubmit{
+							JobBase: prowConfig.JobBase{
+								Name: "ci-kubernetes-unit",
+							},
+						},
+					},
+				},
+				Periodics: []prowConfig.Periodic{
+					{
+						JobBase: prowConfig.JobBase{
+							Name: "ESPv2-continuous-build",
+						},
+					},
+					{
+						JobBase: prowConfig.JobBase{
+							Name: "ci-oss-test-infra-autobump-prow",
+						},
+					},
+					{
+						JobBase: prowConfig.JobBase{
+							Name: "everlast-bump",
+						},
+					},
+				},
+			},
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			a := test.initialJobConfig
+			e := test.expectedJobConfig
+
+			aPer := a.AllPeriodics()
+			ePer := e.AllPeriodics()
+			sortPeriodics(aPer)
+			if len(aPer) != len(ePer) {
+				t.Fatal("Periodic lengths do not match")
+			}
+			for i, actualPer := range aPer {
```

Comment:
> Would `reflect.DeepEqual(expected, actual)` work?

It will for the individual arrays. Updated to use `DeepEqual`.

> Also consider a few test cases to check 1) if jobs sort right, 2) if repositories sort right, and 3) the function doesn't crash if given no jobs.
>
> That may give you a clearer signal for what went wrong without having to maintain complicated "if-then" equality checking.

I subdivided the test cases by `periodics`, `presubmits`, and `postsubmits` as suggested covering cases where **1** jobs sort, **2** repos sort, **3** no crash on empty.